### PR TITLE
feat: handle invalid field names

### PR DESCRIPTION
**Pull request by y-lakhdar**
## Proposed changes

Provide detailed errors and next steps whenever invalid fields are detected by the SDK.

There are essentially 2 possible outcome:

### Outcome 1: invalid fields **can** be normalized
The user can rerun the the same command along with a flag to automatically fix fixable fields.
image

### Outcome 2: invalid fields **cannot** be normalized
The user has to fix his data before running the command
image

## Before merging
- [x] Merge https://github.com/coveo/push-api-client.js/pull/139
- [x] Bump push-api-client version
- [x] Unit Tests:
- [x] Manual Tests:

### CODE REVIEW

**Comment by y-lakhdar** [comment]
Not sure about the verbiage. Is that clear enough?

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:16]: avoid using single letter for your callback.
Yes, it might be less verbose, but is also less clear at first glance.

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:28]: You can use a flatMap to avoid iterating twice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:67]:
```suggestion
    ` ${count} ${pluralized} detected in your data can${
```

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:77]: Unsure, this v.s. `Cli.ux.log(new Array(lines).fill('\n').join(''))` ?

Comment (y-lakhdar) [reply packages/cli/src/lib/push/addCommon.ts:86]: turns out doing that creates the double amount of new lines

File: packages/cli/src/commands/source/push/add.ts
Comment (louis-bompart) [inline packages/cli/src/commands/source/push/add.ts:52]: respect the standard, `withXYZ`

File: packages/cli/src/commands/source/push/add.ts
Comment (louis-bompart) [inline packages/cli/src/commands/source/push/add.ts:109]: You should run the finally if and only if we did successfully affected the status of the source IMO.

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:8]: Can a fieldname ends with a `_` ?

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:20]: You end up going through the array twice, you're probably due to do a method that follows this signature:

```ts
const {fixables, unfixables} = colorizeFields(normalizations);
```

File: packages/cli/src/lib/push/addCommon.ts
Comment (louis-bompart) [inline packages/cli/src/lib/push/addCommon.ts:41]: I think you're not using flatmap properly, you're supposed to return an array at each steps iirc.
> Return a 1-element array to keep the item, a multiple-element array to add items, or a 0-element array to remove the item.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map

**Review by louis-bompart — COMMENTED** [review]
lgtm overall

File: packages/cli/src/commands/source/push/add.ts
Comment (y-lakhdar) [reply packages/cli/src/commands/source/push/add.ts:109]: I understand, but this is a bit overkill I think.
If the first call of `setSourceStatus` returned a 2** status code, it is safe to assume that the second call will work as well.
If the first `setSourceStatus` call fails for some reason, the second will also fail.

Also since the API call responsable to change the status returns `null` as the response, it will be a bit too much to really know wether the CLI did actually change the source status.
We would need 2 API calls to fetch the source status (one at the very beginning and one after the status change).

File: packages/cli/src/lib/push/addCommon.ts
Comment (y-lakhdar) [reply packages/cli/src/lib/push/addCommon.ts:8]: image

File: packages/cli/src/lib/push/addCommon.ts
Comment (y-lakhdar) [reply packages/cli/src/lib/push/addCommon.ts:41]: No. That's the beauty of it. The callback can either return a element of the final array as the element itself or in an array
image

File: packages/cli/src/lib/push/addCommon.ts
Comment (y-lakhdar) [reply packages/cli/src/lib/push/addCommon.ts:20]: true, but I find it harder to read
image

Comment (y-lakhdar) [reply packages/cli/src/lib/push/addCommon.ts:20]: If we had a [partition](https://underscorejs.org/#partition) filter, it would have been cleaner.

Comment (louis-bompart) [reply packages/cli/src/lib/push/addCommon.ts]: You just have to remove the unfixable normalize formating in another func:
https://www.typescriptlang.org/play?ssl=17&ssc=110&pln=17&pc=56#code/C4TwDgpgBACgTgSwHbAIYCMA2ECqSDOArmGAPZzAQAmAYghJlVALxSpIgDcAUAMakFgUfpnIIAXhDoMq+FlAAUAM3qN8ALliIUGbHiIlylWqqoBtALoBKFgD4oAb25RhA-EJUAPXRA1bkaFi4BMRkFNTSjJbyljwu-IJQhEhePn7wAT76oUYRptGssc5QSuSKCe4lplCkSlUy+DZOLi4IdcqmAHQAbqiYCFRNxS1V3kH4nWCE+AAWCg5QnUsqMgA0UEjkALZ9EtSaAOZwEBBIHTKdm3A7-ZKDUAC+VnEtD1AM+NDNI0kpCGPYCZTWbzYY-JadFaMdZXG57KiaUrXVDAABy212dxo5DwqSC50Ylwxt2oVjBLieLwpxQexWOwEIcCQjlGaXWyTxgMe3FpfDcHgxaOJ8OxcFx-x88gUyCmwE07m0BxszHsMsIQmYmqgAHJtVAAPxQKgILYKbUKCBbMCgeqMDaoLYQKzamyaY5UaVIWVWIA

**Review by louis-bompart — APPROVED** [review]
LGTM, if tests passes after merge conflict are resolved

**Review by olamothe — APPROVED** [review]

**Pull request merged**
